### PR TITLE
Log API diffs using ShowAPI and java-diff-utils library.

### PR DESCRIPTION
**Pull request by gkossakowski**
Implement displaying API changes by using textual representation
of an API (ShowAPI) and good, old textual diff algorithm. We are
using java-diff-utils library that is distributed under Apache 2.0
license.

Notice that we have only soft dependency on java-diff-utils. It means
that we'll try to lookup java-diff-utils class through reflection
and fail gracefully if none is found on the classpath. This way
sbt is not getting any new dependency. If user needs to debug
api diffs then it's matter of starting sbt with
`-Dsbt.extraClasspath=path/to/diffutils.jar` option passed
to sbt launcher.

### CODE REVIEW

File: launch/src/main/scala/xsbt/boot/ConfigurationParser.scala [hunk 1]
```diff
@@ -18,7 +18,7 @@ object ConfigurationParser
 	def trim(s: Array[String]) = s.map(_.trim).toList
 	def ids(value: String) = trim(substituteVariables(value).split(",")).filter(isNonEmpty)
 
-	private[this] lazy val VarPattern = Pattern.compile("""\$\{([\w.]+)(\-(.+))?\}""")
+	private[this] lazy val VarPattern = Pattern.compile("""\$\{([\w.]+)(\-(.*))?\}""")
```
Comment (jsuereth) [inline launch/src/main/scala/xsbt/boot/ConfigurationParser.scala:21]: So this just allows empty defaults, right?

Comment (gkossakowski) [inline launch/src/main/scala/xsbt/boot/ConfigurationParser.scala:21]: Yes. It was needed for `resources: ${sbt.extraClasspath-}`.

Comment (gkossakowski) [inline launch/src/main/scala/xsbt/boot/ConfigurationParser.scala:21]: Also, see f426f8ba7f57546331c4003f7b8ed64a2e5f8ea0

I tried to split changes into separate commits :-)

Comment (jsuereth) [inline launch/src/main/scala/xsbt/boot/ConfigurationParser.scala:21]: right, I usually try to consume as a whole first.  Do you mind checking the docs on the launcher to see if they need updated?  

e.g. here https://github.com/sbt/sbt/blob/0.13/src/sphinx/Detailed-Topics/Launcher.rst

Also, would love some docs on the API diff feature as well.  New docs not necessary for this PR, but would like to see them follow soon after.

Comment (gkossakowski) [inline launch/src/main/scala/xsbt/boot/ConfigurationParser.scala:21]: I didn't see anything relevant in launcher docs that would need to be updated but I'll check again.

When it comes to api diff feature then I agree we should have docs on it. I'm happy to do it. I just wanted to get this PR done ASAP as we are under tight deadlines.

Comment (harrah) [inline launch/src/main/scala/xsbt/boot/ConfigurationParser.scala:21]: The relevant section is here: https://github.com/sbt/sbt/blob/0.13/src/sphinx/Detailed-Topics/Launcher.rst#syntax.  It doesn't say the empty string is allowed or disallowed as a default.  I'm not sure it is necessary to add it explicitly.

For the API diff, let's have one line in the changes section (Community/ChangeSummary_0.13.0) under Improvements for now.  A howto section or wherever is appropriate can be added for 0.13.1..

**Comment by gkossakowski** [comment]
@harrah: pushed new commits which address review comments. I can squash those commits if needed.

**Comment by harrah** [comment]
Looks good.  Yes, please squash.  Nothing else needs to be done- it is ready to merge once squashed.

**Comment by gkossakowski** [comment]
@harrah: squashed and pushed.
